test/reg-test/fam-api-reg/fam_backup_restore_reg_test.cpp

```diff
@@ -0,0 +1,195 @@
+/*
+ * fam_put_get_reg_test.cpp
+ * Copyright (c) 2019-2021 Hewlett Packard Enterprise Development, LP. All
+ * rights reserved. Redistribution and use in source and binary forms, with or
+ * without modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ *    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+ * IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * See https://spdx.org/licenses/BSD-3-Clause
+ *
+ */
+#include <fam/fam_exception.h>
+#include <fstream>
+#include <gtest/gtest.h>
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <fam/fam.h>
+
+#include "common/fam_test_config.h"
+
+#define REGION_SIZE 2147483648
+#define DATAITEM_SIZE 1048576
+using namespace std;
+using namespace openfam;
+char *backup_restore_file;
+fam *my_fam;
+Fam_Options fam_opts;
+Fam_Region_Descriptor *desc;
+Fam_Descriptor *item;
+
+// Test case 1 - put get test.
+TEST(FamBackupRestore, BackupSuccess) {
+    Fam_Region_Descriptor *desc;
+    Fam_Descriptor *item;
+    const char *testRegion = get_uniq_str("test", my_fam);
+    const char *firstItem = get_uniq_str("first", my_fam);
+
+    EXPECT_NO_THROW(desc = my_fam->fam_lookup_region(testRegion));
+    EXPECT_NE((void *)NULL, desc);
+
+    // Allocating data items in the created region
+    EXPECT_NO_THROW(item = my_fam->fam_lookup(firstItem, testRegion));
+    EXPECT_NE((void *)NULL, item);
+    char *filename = (char *)malloc(100);
+    try {
+        sprintf(filename, "%s.%s.1", testRegion, firstItem);
+        void *waitobj = my_fam->fam_backup(item, filename);
```

Comment:
If the test is run again without clearing the backup, will the test case fail ?

Reply:
What if we create backup name with timestamp, so that we backup is done all the time ?